experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+            assert isinstance(exp_slice['nrang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify nrang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'frang' in exp_slice.keys()
+            assert isinstance(exp_slice['frang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify frang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'intt' in exp_slice.keys()
+        except AssertionError:
+            try:
+                assert 'intn' in exp_slice.keys()
+            except AssertionError:
+                errmsg = "Slice must specify either an intn or intt"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert isinstance(exp_slice['intn'], int)
+                except AssertionError:
+                    errmsg = "intn must be an integer"
+                    raise ExperimentException(errmsg, exp_slice)
+        else:
+            try:
+                assert isinstance(exp_slice['intt'], int)
+            except AssertionError:
+                errmsg = "intt must be an integer"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert 'intn' in exp_slice.keys()
+                except AssertionError:
+                    pass
+                else:
+                    # TODO Log warning intn will not be used
+                    exp_slice.pop('intn')
+
+        try:
+            assert 'beam_angle' in exp_slice.keys()
+            assert isinstance(exp_slice['beam_angle'], list)
+            for element in exp_slice['beam_angle']:
+                assert isinstance(element, float) or isinstance(element, int) or isinstance(element,
+                                                                                            list)
+                if isinstance(element, list):
+                    for angle in element:
+                        assert isinstance(angle, float) or isinstance(angle, int)
+        except AssertionError:
+            errmsg = "Slice must specify beam_angle that must be a list of floats or lists (of floats)"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'beam_order' in exp_slice.keys()
+            assert isinstance(exp_slice['beam_order'], list)
+            for element in exp_slice['beam_order']:
+                assert isinstance(element, int)
+                assert element < len(exp_slice['beam_angle'])
+        except AssertionError:
+            errmsg = "Slice must specify beam_order that must be a list of ints corresponding to " \
+                     "the order of the angles in the beam_angle list."
+            raise ExperimentException(errmsg, exp_slice)
+
+    @staticmethod
```

Comment:
no. I think it is related to the class so should be included in the class, but does not use anything from the class so that's why I didn't pass in self and added the decorator